docs/pages/api/og.tsx

```diff
@@ -0,0 +1,167 @@
+}
+
+async function loadAssets(): Promise<
+  [
+    { name: string; data: ArrayBuffer; weight: 400 | 700; style: "normal" }[],
+    string
+  ]
+> {
+  const [inter, spaceMono, bg] = await Promise.all([
+    fetch(
+      String(new URL("../../assets/inter-v12-latin-700.ttf", import.meta.url))
+    ).then((res) => res.arrayBuffer()),
+    fetch(
+      String(
+        new URL(
+          "../../assets/space-mono-v12-latin-regular.ttf",
+          import.meta.url
+        )
+      )
+    ).then((res) => res.arrayBuffer()),
+    fetch(String(new URL("../../assets/bg.jpeg", import.meta.url))).then(
+      (res) => res.arrayBuffer()
+    ),
+  ]);
+  return [
+    [
+      {
+        name: "Inter",
+        data: inter,
+        weight: 700 as const,
+        style: "normal" as const,
+      },
+      {
+        name: "Space Mono",
+        data: spaceMono,
+        weight: 400 as const,
+        style: "normal" as const,
+      },
+    ],
+    _arrayBufferToBase64(bg),
+  ];
+}
+
+export const config = {
+  runtime: "experimental-edge",
+};
+
+export default async function openGraphImage(
+  req: NextApiRequest
+): Promise<ImageResponse> {
+  const [fonts, bg] = await loadAssets();
+  try {
+    const { searchParams } = new URL(req.url);
+
+    const type = searchParams.get("type");
+
+    // ?title=<title>
+    const hasTitle = searchParams.has("title");
+    const title = hasTitle
+      ? searchParams.get("title")?.slice(0, 100)
+      : type === "pack"
+      ? "The successor to Webpack"
+      : type === "repo"
+      ? "The build system that makes ship happen"
+      : "";
+
+    return new ImageResponse(createElement(OGImage, { title, type, bg }), {
+      width: 1200,
+      height: 630,
+      fonts,
+    });
+  } catch (e: unknown) {
+    if (e instanceof Error) {
+      return new Response(`Failed to generate the image: ${e.message}`, {
+        status: 500,
+      });
+    }
+    return new Response("Failed to generate image", { status: 500 });
+  }
+}
+
+export function OGImage({
+  title,
+  type,
+  bg,
+}: {
+  title: string;
+  type: string;
+  bg: string;
+}): JSX.Element {
+  return (
+    <div
+      style={{
+        display: "flex",
+        flexDirection: "column",
+        alignItems: "center",
+        justifyContent: "center",
+        width: "100%",
+        height: "100%",
+        fontFamily: "Inter",
+        fontWeight: 700,
+        fontSize: 60,
+        backgroundImage: `url(data:image/jpeg;base64,${bg})`,
+        backgroundSize: "1200px 630px",
+        color: "#fff",
+      }}
+    >
+      {/* eslint-disable-next-line  @next/next/no-img-element, jsx-a11y/alt-text */}
+      <div style={{ display: "flex", height: 97 * 1.1, alignItems: "center" }}>
+        {type === "pack" ? (
+          <PackLogo height={96 * 1.1} width={594 * 1.1} />
+        ) : type === "repo" ? (
+          <RepoLogo height={83 * 1.1} width={616 * 1.1} />
+        ) : (
+          <TurboLogo height={97 * 1.1} width={459 * 1.1} />
```

Comment:
Some notes on where these magic numbers are coming from would be useful

Reply:
I have _no_ idea. They're pulled from the existing implementation.